### PR TITLE
Fix: prevent stalled migration issue

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

Migration should complete when the plugin users upgrade from GiveWP <=2.0.  Rick provided me with a database that has an issue and after investigation, I find out that the migration process was blocked by a migration that has a dependency migration that was removed earlier. I added a fix to complete the migration.

Slack discussion: https://lw.slack.com/archives/C02PQ6F648P/p1654813134457039

## Testing Instructions

You will find out that database dump in the slack channel. You should be able to complete the migration.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by ravinderk** [comment]
> @ravinderk Should this be checking for something to make sure that the upgrade did in fact happen?

@JasonTheAdams It does not need to check for anything.
1. `v20_logs_upgrades ` legacy migration has been removed in the recent version of GiveWP, so completing it by default with auto-running migration will not affect anything but it will unblock `v20_rename_donor_tables ` migration if it was struck.
2. `MigrateExistingLogs` has been fixed. it will complete itself if the log table does not exist instead of continuing running till infinite time.

File: includes/admin/upgrades/upgrade-functions.php [hunk 1]
```diff
@@ -337,7 +337,6 @@ function give_show_upgrade_notices( $give_updates ) {
 			'callback' => 'give_v20_rename_donor_tables_callback',
 			'depend'   => [
 				'v20_move_metadata_into_new_table',
-				'v20_logs_upgrades',
```
Comment (JasonTheAdams) [inline includes/admin/upgrades/upgrade-functions.php:340]: Just making sure I think through this. If we're removing the dependency, is it still necessary to mark it as complete in a migration?

Comment (ravinderk) [reply includes/admin/upgrades/upgrade-functions.php:340]: @JasonTheAdams is required. We store new migration registration data in the options table (legacy logic for background update) and then use that data to process migration, so removing `v20_logs_upgrades ` migration in code will fix the issue for plugin users who will upgrade from Give<=2.0 in the future but for the existing user, we have to mark `v20_logs_upgrades` migration complete in the database to resume migration process.

**Review by rickalday — APPROVED** [review]
This PR cleared the stalled database upgrade. I tried the DB healthcheck plugin to see it that copied over the data from the paymentmeta table over to the donationmeta table but that didn’t work.

**Comment by JasonTheAdams** [comment]
I'm going to leave this for now as it didn't fully resolve the issue. We'll get this out in the next release.

**Review by rickalday — APPROVED** [review]
All is good now. I tested this morning with Ravinder and this fixes stalled database issue.  The master branch of the DB healthcheck plugin was used to fix the other remaining issue for one particular user.

**Pull request merged**